src/command_frame.cc

```diff
@@ -1,71 +1,112 @@
-
 #include <cstdlib>  // EXIT_SUCCESS, EXIT_FAILURE
 #include <iostream> // std::cout
+#include <sstream>  // std::ostringstream
 
 #include "command.h"
 #include "utils.h"
 
-auto intelligence::jsonschema::cli::frame(
-    const std::span<const std::string> &arguments) -> int {
-  const auto options{parse_options(arguments, {})};
-  CLI_ENSURE(!options.at("").empty(), "You must pass a JSON Schema as input")
-  const sourcemeta::jsontoolkit::JSON schema{
-      sourcemeta::jsontoolkit::from_file(options.at("").front())};
-
-  sourcemeta::jsontoolkit::ReferenceFrame frame;
-  sourcemeta::jsontoolkit::ReferenceMap references;
-  sourcemeta::jsontoolkit::frame(schema, frame, references,
-                                 sourcemeta::jsontoolkit::default_schema_walker,
-                                 resolver(options))
-      .wait();
-
-  for (const auto &[key, entry] : frame) {
-    std::cout << "(LOCATION) URI: ";
-    std::cout << key.second << "\n";
-    std::cout << "    Schema           : " << entry.root.value_or("<ANONYMOUS>")
-              << "\n";
-    std::cout << "    Pointer          :";
-    if (!entry.pointer.empty()) {
-      std::cout << " ";
+const char* enumToString(sourcemeta::jsontoolkit::ReferenceEntryType type) {
+    switch(type) {
+        case sourcemeta::jsontoolkit::ReferenceEntryType::Resource: return "Resource";
+        case sourcemeta::jsontoolkit::ReferenceEntryType::Anchor: return "Anchor";
+        case sourcemeta::jsontoolkit::ReferenceEntryType::Pointer: return "Pointer";
+        default: return "Unknown";
     }
+}
 
-    sourcemeta::jsontoolkit::stringify(entry.pointer, std::cout);
-    std::cout << "\n";
-    std::cout << "    Base URI         : " << entry.base << "\n";
-    std::cout << "    Relative Pointer :";
-    if (!entry.relative_pointer.empty()) {
-      std::cout << " ";
-    }
+auto intelligence::jsonschema::cli::frame(const std::span<const std::string> &arguments) -> int {
+    const auto options{parse_options(arguments, {"json", "j"})};
+    CLI_ENSURE(!options.at("").empty(), "You must pass a JSON Schema as input")
+    const sourcemeta::jsontoolkit::JSON schema{
+        sourcemeta::jsontoolkit::from_file(options.at("").front())
+    };
 
-    sourcemeta::jsontoolkit::stringify(entry.relative_pointer, std::cout);
-    std::cout << "\n";
-    std::cout << "    Dialect          : " << entry.dialect << "\n";
-  }
+    sourcemeta::jsontoolkit::ReferenceFrame frame;
+    sourcemeta::jsontoolkit::ReferenceMap references;
+    sourcemeta::jsontoolkit::frame(schema, frame, references,
+                                   sourcemeta::jsontoolkit::default_schema_walker,
+                                   resolver(options))
+        .wait();
 
-  for (const auto &[pointer, entry] : references) {
-    std::cout << "(REFERENCE) URI: ";
-    sourcemeta::jsontoolkit::stringify(pointer.second, std::cout);
-    std::cout << "\n";
+    bool outputJson = options.contains("json") || options.contains("j");
```

Comment:
All variables must be camel_case

Reply:
Also, try to use `const` whenever possible. This variable is not supposed to change, so it should be a constant